### PR TITLE
feat(spannerio): add CDC metadata model for database table

### PR DESCRIPTION
**Pull request by zoercai**

### CODE REVIEW

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/Metadata.java [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+        inclusiveStart = true;
+      }
+      if (inclusiveEnd == null) {
+        inclusiveEnd = false;
+      }
+      return new Metadata(partitionToken, parentToken, startTimestamp, inclusiveStart, endTimestamp,
+          inclusiveEnd, heartbeatSeconds, state, createdAt, updatedAt);
+    }
+  }
+}
```
Comment (zoercai) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/Metadata.java:242]: add newline

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/Metadata.java [hunk 2]
```diff
@@ -0,0 +1,242 @@
+package org.apache.beam.sdk.io.gcp.spanner.cdc;
+
+import com.google.cloud.Timestamp;
+import com.google.common.base.Objects;
+
+public class Metadata {
```
Comment (thiagotnunes) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/Metadata.java:6]: Could you change the name of the class to reflect the name of the table? Something like "PartitionMetadata" maybe?

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/Metadata.java [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+      this.createdAt = createdAt;
+      return this;
+    }
+
+    public Builder setUpdatedAt(Timestamp updatedAt) {
+      this.updatedAt = updatedAt;
+      return this;
+    }
+
+    public Metadata build() {
```
Comment (thiagotnunes) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/Metadata.java:231]: In the next PR, could you create a unit test for this?

Comment (zoercai) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/Metadata.java]: will do!

**Review by thiagotnunes — APPROVED** [review]
Thanks Zoe, could you please change the title of the PR to follow https://www.conventionalcommits.org/en/v1.0.0/?

Something like `feat(spannerio): adds cdc metadata model` maybe?

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/spanner/cdc/PartitionMetadataTest.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        .setPartitionToken(A_PARTITION_TOKEN)
+        .setParentToken(A_PARTITION_TOKEN)
+        .setStartTimestamp(A_TIMESTAMP)
+        .setEndTimestamp(A_TIMESTAMP)
+        .setHeartbeatSeconds(10)
+        .setState(State.CREATED)
+        .setCreatedAt(A_TIMESTAMP)
+        .setUpdatedAt(A_TIMESTAMP)
+        .build();
+    assertTrue(partitionMetadata.isInclusiveStart());
```
Comment (thiagotnunes) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/spanner/cdc/PartitionMetadataTest.java:31]: Could we test the whole object?

```
assertEquals(new PartitionMetadata(...), partitionMetadata)
```

Comment (zoercai) [reply sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/spanner/cdc/PartitionMetadataTest.java]: done :)

**Review by thiagotnunes — APPROVED** [review]
LGTM

**Review by hengfengli — APPROVED** [review]
LGTM.

**Pull request merged**